base/missing.jl

```diff
@@ -19,35 +19,35 @@ showerror(io::IO, ex::MissingException) =
     print(io, "MissingException: ", ex.msg)
 
 nonmissingtype(::Type{T}) where {T} = Core.Compiler.typesubtract(T, Missing)
+function nonmissingtype_checked(T::Type)
+    R = nonmissingtype(T)
+    R >: T && throw(MissingException("could not compute non-missing type"))
```

Comment:
We probably shouldn't use `MissingException` for this: until now it's reserved for situations where a `missing` value appears in a place where it's not allowed. This sounds more like an `ArgumentError` or a generic `ErrorException` to me.
```suggestion
    R >: T && throw(ArgumentError("could not compute non-missing type from $T"))
```